scorep/instrumenters/scorep_instrumenter_base.py

```diff
@@ -0,0 +1,157 @@
+import abc
+import inspect
+import os
+from scorep.instrumenters import base_instrumenter
+
+
+class ScorepInstrumenterBase(base_instrumenter.BaseInstrumenter):
+    """Base class for all instrumenters using Score-P"""
+
+    def __init__(self, scorep_bindings, enable_instrumenter=True):
+        """
+        @param enable_instrumenter true if the tracing shall be initialised.
+            Please note, that it is still possible to enable the tracing later using register()
+        """
+        self._scorep_bindings = scorep_bindings
+        self._tracer_registered = False
+        self._enabled = enable_instrumenter
+
+    @abc.abstractmethod
+    def _enable_instrumenter(self):
+        """Actually enable this instrumenter to collect events"""
+
+    @abc.abstractmethod
+    def _disable_instrumenter(self):
+        """Stop this instrumenter from collecting events"""
+
+    def register(self):
+        """Register this instrumenter (collect events)"""
+        if not self._tracer_registered:
+            self._enable_instrumenter()
+            self._tracer_registered = True
+
+    def unregister(self):
+        """Unregister this instrumenter (stop collecting events)"""
+        if self._tracer_registered:
+            self._disable_instrumenter()
+            self._tracer_registered = False
+
+    def get_registered(self):
+        """Return whether this instrumenter is currently collecting events"""
+        return self._tracer_registered
+
+    def run(self, cmd):
+        """
+        Run the compiled command.
+        Registers this instrumenter for the duration of the command and unregisters it afterwards
+        """
+        self.runctx(cmd)
+
+    def runctx(self, cmd, globals=None, locals=None):
+        """
+        Run the compiled command.
+        Registers this instrumenter for the duration of the command and unregisters it afterwards
```

Comment:
> Registers this instrumenter for the duration of the command and unregisters it afterwards

only if it is supposed to, e.g. if `enable_instrumenter` was true during init.

Reply:
Done: 5a192ce